### ISSUE TITLE
Is it possible to get `select` logprobs like in versions <0.1.0?

### ISSUE THREAD

**Issue by wjn0** [guidance-ai/guidance#590]
**Is your feature request related to a problem? Please describe.**
In old (handlebar) versions one could do something like:

```python
program = guidance("The quick brown fox jumps over the lazy {{select 'animal' options=valid_animals logprobs='animals_logprobs'}}")
returned = program()
print(returned["animals_logprobs"])
```

to get the relative logprobs of each option. Is that possible?

**Describe the solution you'd like**

Something similar to the above maybe.

**Describe alternatives you've considered**

`llm._cache_state["logits"]` is not the same thing -- only provides next-token logits, as I understand it.

**Comment by wjn0** [comment]
Ah, just found this comment: https://github.com/guidance-ai/guidance/blob/cf355c7ac12ce7ce9ddddea115329d7ec9eeb939/guidance/_grammar.py#L489C6-L489C6

- so maybe not.

I would be interested in implementing this if the owners are amenable, but would maybe just need a little bit of guidance (ha) to better understand the repo and whether or not the <0.1 implementation should just be pulled forward or whether something new needs to happen.

**Comment by talglobus** [comment]
Happy to help implement this
